pkgs/tools/package-management/nix-serve/default.nix

```diff
@@ -12,20 +20,28 @@ stdenv.mkDerivation rec {
     sha256 = "0mjzsiknln3isdri9004wwjjjpak5fj8ncizyncf5jv7g4m4q1pj";
   };
 
-  buildInputs = [ perl nix ]
-    ++ (with perlPackages; [ DBI DBDSQLite Plack Starman ]);
+  propagatedBuildInputs = [
+    nix
+    perl
+    perlPackages.DBDSQLite
+    perlPackages.DBI
+    perlPackages.Plack
+    perlPackages.Starman
+  ];
 
-  dontBuild = false;
+  phases = ["unpackPhase" "installPhase"];
 
-  # FIXME: unfortunate cut&paste.
   installPhase = ''
     mkdir -p $out/libexec/nix-serve
     cp nix-serve.psgi $out/libexec/nix-serve/nix-serve.psgi
 
     mkdir -p $out/bin
     cat > $out/bin/nix-serve <<EOF
     #! ${stdenv.shell}
-    PERL5LIB=$PERL5LIB exec ${perlPackages.Starman}/bin/starman $out/libexec/nix-serve/nix-serve.psgi "\$@"
+    export PATH=${nix.out}/bin:\$PATH
```

Comment:
Why would `bzip2.out` not be what we want? I've never used `makeBinPath` before; I tried it in the repl and it works, but I'm not sure why it would be an improvement over a straightforward `/bin`...